source/fuzz/fact_manager.cpp

```diff
@@ -761,8 +756,13 @@ void FactManager::DataSynonymAndIdEquationFacts::
     std::vector<const protobufs::DataDescriptor*> convert_u_to_f_lhs;
 
     for (const auto& fact : id_equations_) {
+      if (!context->get_def_use_mgr()->GetDef(fact.first->object())) {
+        // Skip non-existent instructions.
+        continue;
+      }
```

Comment:
This might be too extreme. We could mark two data descriptors as synonymous even if one of them doesn't exist in the module. We can do that if elements of equivalence classes of both data descriptors are guaranteed to have compatible types.

This can be useful since some of the elements of those equivalence classes might still exist in the module.

A drawback of this approach, though, is that it might be harder to manually analyze a shader in this case.

Anyway, we can't do this ATM since an assertion fails in
https://github.com/KhronosGroup/SPIRV-Tools/blob/4dd122392f3ad757e70951a1198479bf233d4cd8/source/fuzz/fact_manager.cpp#L720-L723

Reply:
@afd, @paulthomson, do you think we should add synonyms between non-existent data descriptors?